build.sh

```diff
@@ -5,11 +5,156 @@
 # This script is basically automation script of dev.md
 # It is supposed to run in same folder where the file lives (root).
 
-# 1. enter into build folder
-mkdir -p build && cd build
```

Comment:
do we want to enforce all below installations happening in /build folder, otherwise some downloads may pollute source tree?

Reply:
we can get nebula root by this command 
`ROOT=$(git rev-parse --show-toplevel)`

Reply:
and then we can do 
`
BUILD_DIR=$ROOT/build
mkdir -p $BUILD_DIR && cd $BUILD_DIR
`

Reply:
got you, thanks!